crates/starknet-devnet-core/src/state/mod.rs

```diff
@@ -113,538 +244,253 @@ impl blockifier::state::state_api::StateReader for DevnetState {
     ) -> blockifier::state::state_api::StateResult<
         blockifier::execution::contract_class::ContractClass,
     > {
-        let contract_class =
-            crate::traits::DevnetStateReader::contract_class_at(self, &class_hash.0.into())
-                .map_err(|_| {
-                    blockifier::state::errors::StateError::UndeclaredClassHash(class_hash)
-                })?;
-
-        blockifier::execution::contract_class::ContractClass::try_from(contract_class)
-            .map_err(|err| blockifier::state::errors::StateError::StateReadError(err.to_string()))
+        self.state.get_compiled_contract_class(class_hash)
     }
 
     fn get_compiled_class_hash(
         &mut self,
         class_hash: starknet_api::core::ClassHash,
     ) -> blockifier::state::state_api::StateResult<starknet_api::core::CompiledClassHash> {
-        let compiled_class_hash =
-            crate::traits::DevnetStateReader::compiled_class_hash_at(self, &(class_hash.0.into()));
-        Ok(starknet_api::core::CompiledClassHash(compiled_class_hash.into()))
+        self.state.get_compiled_class_hash(class_hash)
     }
 }
 
-impl StarknetState {
-    /// this method clears the state from data that was accumulated in the StateCache
-    /// and restores it to the data in the state_reader, which is the "persistent" data
-    pub(crate) fn clear_dirty_state(&mut self) {
-        self.state = CachedState::new(
-            self.state.state.clone(),
-            GlobalContractCache::new(GLOBAL_CONTRACT_CACHE_SIZE_FOR_TEST),
-        );
+impl CustomStateReader for StarknetState {
+    fn is_contract_deployed(&mut self, contract_address: ContractAddress) -> bool {
+        let api_address = contract_address.try_into().unwrap();
```

Comment:
While looking into the contract deployment assertion, I realized there's this `try_into().unwrap()` which I had left to think about later. Is that an acceptable one?

Reply:
I missed it, but not acceptable

Reply:
Changed in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/329/commits/ac7329ee50833d18eaf23ff50ee8061a648bc0ff